### DISCUSSION TITLE
Change data variable attribute in NetCDF file stored in s3 bucket?

### DISCUSSION THREAD

**Discussion by mliukis** [pydata/xarray#9903] [Q&A] [answered]
Hello,
I know there is a way to change a value of the attribute in-place for the data variable in Zarr which resides in s3 bucket (without storing modified data set to the local Zarr, then copying back to the s3 bucket).

Is there a way to do the same for the NetCDF file stored in s3 bucket?

Thanks in advance!
Masha

**Comment by jhamman** [comment, marked as answer]
Good question. Unfortunately, because S3 does not act like file system, this is not possible. Objects can not be updated incrementally. So any updates to a NetCDF file will require rewriting the entire object.

**Reply by mliukis** [reply]
Thank you!
